asserts/database.go

```diff
@@ -108,3 +123,53 @@ func (db *Database) ImportKey(authorityID string, privKey *packet.PrivateKey) (f
 	}
 	return fingerp, nil
 }
+
+// use a generalized matching style along what PGP does where keys can be
+// retrieved by giving suffixes of their fingerprint,
+// for safety suffix must be at least 64 bits though
+// TODO: may need more details about the kind of key we are looking for
+func (db *Database) findPublicKeys(authorityID, fingerprintSuffix string) []PublicKey {
+	suffixLen := len(fingerprintSuffix)
+	if suffixLen%2 == 1 {
+		panic(fmt.Errorf("findPublicKeys: fingerprintSuffix cannot specify a half byte"))
```

Comment:
Sounds like these (here and below) shouldn't panic.. the fingerprint is coming inside the signature, which is very much untrusted before validated, so panicking here may conceivably mean a third party crashing a snappy system by handing it an invalid assertion, with a broken fingerprint.